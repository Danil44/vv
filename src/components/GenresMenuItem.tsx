import React from 'react';
import { jsx, css } from '@emotion/core';
import { lightGrey, mainOrange } from '../colors';
import { ReactComponent as RightArrowIcon } from '../images/right-arrow.svg';
/**@jsx jsx */

type GenresMenuItemProps = {
  active?: boolean;
  name: string;
};

export default function GenresMenuItem({
  active = false,
  name,
}: GenresMenuItemProps) {
  return (
    <li
      css={css`
        color: ${active ? '#fff' : lightGrey};
        padding: 1em 0 1em 0.938em;
        border: 2px solid ${active ? mainOrange : 'transparent'};
        border-radius: 6px;
        font-size: 1em;
        display: flex;
        align-items: center;
      `}
    >
      <span
        css={css`
          margin-right: 5px;
          text-transform: capitalize;
        `}
      >
        {name}
      </span>
      {active && <RightArrowIcon width="10px" height="10px" />}
    </li>
  );
}
