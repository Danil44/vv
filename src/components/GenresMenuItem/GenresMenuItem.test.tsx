import React from 'react';
import { ShallowWrapper, shallow, mount, ReactWrapper } from 'enzyme';
import GenresMenuItem from './GenresMenuItem';
import { ReactComponent as RightArrowIcon } from '../../images/right-arrow.svg';

describe('<GenresMenuItem />', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<GenresMenuItem active={false} name="" />);
  });

  it('has not changed', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the genre name', () => {
    wrapper.setProps({ name: 'Comedy' });

    expect(wrapper.find('span').text()).toBe('Comedy');
  });

  it('should render RightArrow icon component if active prop is set to true', () => {
    wrapper.setProps({ active: true });

    expect(wrapper.containsMatchingElement(<RightArrowIcon />)).toEqual(true);
  });
});
