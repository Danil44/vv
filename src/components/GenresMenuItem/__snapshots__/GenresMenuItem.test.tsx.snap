// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenresMenuItem /> has not changed 1`] = `"<li class=\\"css-1d8muia-GenresMenuItem\\"><span class=\\"css-2opb7a-GenresMenuItem\\"></span></li>"`;
